tycho-apitools-plugin/src/main/java/org/eclipse/tycho/apitools/ApiAnalysisMojo.java

```diff
@@ -0,0 +1,176 @@
+import org.eclipse.tycho.p2.resolver.facade.P2ResolutionResult;
+import org.eclipse.tycho.p2.resolver.facade.P2Resolver;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolverFactory;
+import org.eclipse.tycho.p2.target.facade.TargetPlatformConfigurationStub;
+
+/**
+ * Performs a PDE-API Tools analysis of this project.
+ */
+@Mojo(name = "analyse", defaultPhase = LifecyclePhase.VERIFY, threadSafe = true, requiresDependencyCollection = ResolutionScope.COMPILE_PLUS_RUNTIME, requiresOnline = true)
+// TODO: or later?
+public class ApiAnalysisMojo extends AbstractMojo {
+
+	@Component(hint = TychoServiceFactory.HINT)
+	private EquinoxServiceFactory equinox;
+
+	@Requirement(hint = "connect")
+	private EquinoxServiceFactory serviceFactory;
+
+	@Component
+	private Logger plexusLogger;
+
+	@Parameter(property = "project", readonly = true)
+	private MavenProject project;
+
+	@Component(role = TychoProject.class)
+	private Map<String, TychoProject> projectTypes;
+
+	@Parameter(property = "tycho.apitools.analyse.baseline")
+	private List<URL> baselines; // FIXME: make required? or use String? Just like CompareWithBaselineMojo?
+
+	@Parameter(defaultValue = "false", property = "tycho.apitools.analyse.skip")
+	private boolean skip;
+
+	@Component(role = APIAnalyser.class)
+	private APIAnalyser apiAnalyser;
+
+	@Override
+	public void execute() throws MojoExecutionException, MojoFailureException {
+		if (skip) {
+			getLog().info("Skipped");
+			return;
+		}
+		if (new File(project.getBasedir(), JarFile.MANIFEST_NAME).isFile()) { // TODO: consider non PDE bundles
+
+			File baselineFile2;
+			try {
+				baselineFile2 = getBaselineFile();
+			} catch (URISyntaxException e) {
+				throw new MojoExecutionException("Failed to fetch baseline file", e);
+			}
+			System.out.println("Baseline file: " + baselineFile2);
+
+			synchronized (ApiAnalysisMojo.class) { // TODO: ensure thread-safe
+
+				ITargetPlatformService service = equinox.getService(ITargetPlatformService.class);
+
+				if (serviceFactory != null) { // FIXME: serviceFactory is always null
+					APIAnalyser service2 = serviceFactory.getService(APIAnalyser.class);
+					System.out.println(service2);
+				}
+				File projectFile = project.getArtifact().getFile();
+				Set<File> projectDependencies = collectProjectDependencyPaths();
+
+				IApiProblem[] problems;
+				try {
+					problems = apiAnalyser.analyse(projectFile, projectDependencies, baselineFile2);
+				} catch (IOException | CoreException e) {
+					throw new MojoExecutionException("Failed to analyse API", e);
+				}
+				for (IApiProblem problem : problems) {
+					if (problem.getSeverity() == ApiPlugin.ERROR) {
+						// TODO: collect problems and make it simple to parse it like in latest API-app
+						// improvements
+						throw new MojoFailureException("API-problems:" + problem.getMessage());
+					}
+				}
+			}
+		}
+	}
+
+	@SuppressWarnings("removal")
+	public Set<File> collectProjectDependencyPaths() {
```

Comment:
Actually I think it might be even better to actually inspect the compile path of the maven project here but we could investigate this later on...